### DISCUSSION TITLE
New experimental asset graph improvements

### DISCUSSION THREAD

**Discussion by braunjj** [dagster-io/dagster#16657] [Ideas]
We're working on a few new enhancements to the asset graph that we're aiming to ship in the next major release. You can enable these from the Experimental Features toggle in your user settings in Dagster 1.4.15 or later.

### **Faster DAG rendering:** 
We're experimenting with a new layout algorithm that renders large graphs more quickly.

### **New Asset sidebar explorer**
The new sidebar for the lineage graph renders a flattened view of the entire asset lineage tree. You can use this new sidebar to quickly jump to a specific asset, navigate to different parts of the graph, and even filter the graph by selecting "show downstream" or "show upstream" from the dropdown menu.

![image](https://github.com/dagster-io/dagster/assets/2798333/12fc8d20-9051-47d4-9b55-7a43b1b1a712)

### **Asset groups are now collapsable**
You can now expand and collapse any group in the graph

https://github.com/dagster-io/dagster/assets/2798333/838ad8c4-cd1f-45b3-8d2b-3eca1a4b79d0

Give it a try and let us know you think!

**Comment by malow106** [comment]
Hey, I love the idea ! I think I will use it often :) 
On my side the two buttons for selecting up/dowstream dependencies don't work. The generated expression `"T024D"*` seems to not consider key prefix. In my case the correct expression should be :  `"ETL/T024D"*`

**Comment by jasonicarter** [comment]
Love it! And much easier to navigate for our dbt project than the normal docs, plus being able to select/query up and down stream dependencies and then materialize them is perfect.

**Comment by stevenmurphy12** [comment]
It's looking good, I noticed the changes being referred to in the 1.5.9 release. A couple of thoughts:

1.

With the experimental experience being enabled, I see roughly the same thing being displayed twice: code location, which can then be expanded to show asset groups.

![image](https://github.com/dagster-io/dagster/assets/6832968/41d96501-37ad-46ab-ae85-db2d2f96bb41)

2.

A 'folder' icon represents a code location, which is helpful for navigation and organisational purposes. A 'group of groups' if you will.

My code is split into multiple code locations in order to get this organisation, but it introduces more overhead in the background as a result. I'm having to create many code locations when I don't really need it, apart from the UI organisation it offers. Would it be possible to add a feature by which we define our own 'group of groups', as opposed to relying on using code locations to achieve this? 

(more detail can be found here - https://github.com/dagster-io/dagster/discussions/17949)

**Reply by braunjj** [reply]
Thanks for the feedback @stevenmurphy12! We are actively exploring improvements to our grouping concepts and we are planning to make asset groups hierarchical so you will be able to create groups of groups. This will ship as a separate workstream from these asset graph improvements though. 

We are also planning some changes to the information architecture and nav to eliminate the double sidebar you show in your screenshot.

**Reply by stevenmurphy12** [reply]
Appreciate the info, thank you 👍

**Comment by EtienneT** [comment]
I like the faster rendering speed, but what I liked about the older graph was that you could see the whole graph all at once, viewing the state of the materializations.  Now you have to open the groups one by one to view it.  A simple expand/collapse all would solve this problem.  Also it would be nice to retain the collapsed/expanded state somehow (cookie, localStorage or indexedDb in the browser or in the dagster db) so that the next time you open the asset graph, you don't have to adjust those.

**Reply by braunjj** [reply]
Thanks for the feedback @EtienneT. There is actually an expand/collapse all button down in the bottom right, but we can do more to make this more clear. Storing the group state in localstorage is a great idea too. Expect to see that added soon.

Lastly we're planning to add status summaries to each group node. So in the future, you'll be able to see the overall status of each asset without having to click into each group (see the screenshot below).

Hope this helps! 
<img width="380" alt="image" src="https://github.com/dagster-io/dagster/assets/2798333/375c3106-e66c-4791-acba-325bf3a9a3cc">

**Reply by EtienneT** [reply]
Excellent!  I did not see the expand all button.  Works great.  One quick comment, when clicking expand all it would be nice to adjust the zoom level automatically to show the whole graph.

**Reply by tacastillo** [reply]
Hello! Just here to provide an anecdote. TIL about the "collapse/expand all" button.

**Comment by EtienneT** [comment]
The scroll wheel of the mouse should control the zoom level just like in google maps.

**Reply by salazarm** [reply]
Hi EtienneT, this used to be the default behavior but we found that zooming is not as common as panning around so to zoom you should hold shift while scrolling the wheel of the mouse.
